python/cudf/cudf/_cuda/gpu.pyx

```diff
@@ -23,20 +23,23 @@ from cudf._cuda.gpu cimport underlying_type_attribute as c_attr
 class CUDARuntimeError(RuntimeError):
 
     def __init__(self, cudaError_t status):
-        self.status = status
         cdef str name = cudaGetErrorName(status).decode()
         cdef str msg = cudaGetErrorString(status).decode()
         super(CUDARuntimeError, self).__init__(
             '%s: %s' % (name, msg))
 
+        self.status = status
+
     def __reduce__(self):
         return (type(self), (self.status,))
 
 
 class CUDADriverError(RuntimeError):
 
     def __init__(self, CUresult status):
-        self.status = status
+        cdef CUresult init_status = cuInit(0)
+        if init_status != 0:
+            raise CUDADriverError(init_status)
```

Comment:
So this is probably not a viable way to handle this error as we could wind up in infinite recursion. It is possible `cuInit` would just be kicking up some error caused by an unrelated operation. However more likely if this is failing there is some more fundamental configuration issue. Not sure how we want to handle that. One option would be just to raise a `RuntimeError` with some message and the status. Another might be to check for the few error status here and handle them accordingly. There may be other options.